usb/esp_tinyusb/include_private/tinyusb_ext_phy.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-FileCopyrightText: 2020-2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "esp_err.h"
+#include "esp_private/usb_phy.h"
+#include "soc/usb_pins.h"
+#include "tinyusb.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Configure and install new USB PHY
+ *
+ * @param[in] config tinyusb stack specific configuration
+ */
+esp_err_t tinyusb_ext_phy_new(const tinyusb_config_t *config);
```

Comment:
Will this API be extensible in the future to support ULPI PHYs for the device stack?

Reply:
@Dazza0 Probably, yes. I don't know the requirements for the ULPI PHYs, but right now it doesn't matter.
The only difference after this PR is **where** do we need to do this changes to support ULPI PHY: in `tinyusb.c` (current version) or in `tinyusb_ext_phy.c` (this PR)

Reply:
The changes will need to be in the `usb_phy.h` API. The `usb_new_phy()` function will need to...

- Select which controller to connect to a PHY (e.g., USB HS, USB FS, USJ)
- Select which PHY interface to use for this controller (FSLS vs UTMI vs ULPI). Note that each controller may only support a subset of these interfaces.
- Whether this interface routes to an internal PHY, or an external PHY. If eternal, users should provide the list of GPIOs that map to each signal of the interface.

The current `usb_phy_config_t` doesn't support the API model described above (thus the need to refactor `usb_phy.h` API.